parsers/marburg.py

```diff
@@ -1,61 +1,73 @@
-    # additional charges
-    prices = {}
-    for p in document.find_all('p'):
-        match = employeePrice.search(p.text)
-        if match:
-            prices['employee'] = match.group('price')
-        match = otherPrice.search(p.text)
-        if match:
-            prices['other'] = match.group('price')
-    if len(prices) != 2:
-        print('Could not extract addtional charges for employee and others')
-    canteen.setAdditionalCharges('student', prices)
-    # find
-    mensa_data = document.find('h1', text=re.compile(mensa)).parent
-    while type(mensa_data) != Tag or mensa_data.name != 'div'\
-            or 'tx-cagcafeteria-pi1' not in mensa_data.get('class', []):
-        mensa_data = mensa_data.next_sibling
-    weekDays = extractWeekDates(mensa_data.find('h2').text)
-    for day_headline in mensa_data.find_all('h3'):
-        date = weekDays[day_headline.text]
-        day_table = day_headline.next_sibling.next_sibling
-        for tr_menu in day_table.tbody.find_all('tr'):
-            category = tr_menu.find_all('td')[0].text.strip()
-            name = tr_menu.find_all('td')[1].text.replace('\r\n', ' ').strip()
-            canteen.addMeal(date, category, name, [], tr_menu.find_all('td')[2].text)
-
-
-def parse_url(url, mensa, *weeks, today):
+from pyopenmensa.feed import LazyBuilder, extractDate
+
+heading_regex = re.compile(r'^h[1-6]$')
+notes_regex = re.compile(r'\((\d+,?\s*)*\)\s*')
+price_regex = re.compile(r'(?P<price>\d+[,.]\d{2}) ?€')
+
+roles = ('student',)
+
+
+def parse_dish(dish, canteen):
+    date = extractDate(dish['data-date'])
+
+    name = dish.find(class_='neo-menu-single-title')
+    try:
+        notes = set(x['title'] for x in name.find_all(name='abbr'))
+    except AttributeError:
+        # No Name
+        return
+    name = re.sub(notes_regex, '', name.get_text(strip=True))
+
+    try:
+        category = dish.find(class_='neo-menu-single-type').text
+    except AttributeError:
+        # No categprie this is maby a side
+        category = 'Beilagen: ' + dish.find_previous(name='h2').text.capitalize()
+
+    try:
+        price = dish.find(class_='neo-menu-single-price').text
+        prices = price_regex.findall(price)
+    except AttributeError:
+        # No Price
+        prices = []
+
+    canteen.addMeal(date, category, name, notes, prices, roles)
+    return
+
+
+def parse_week(url, canteen, data_canteen):
+    data = urlopen(url).read().decode('utf-8')
+    document = parse(data, 'lxml')
+    try:
+        dishes = document.find(class_='neo-menu-single-dishes').find_all(name='tr', attrs={"data-canteen": data_canteen})
+    except AttributeError:
+        # No disches
+        pass
+    try:
+        sides = document.find(class_='neo-menu-single-modals').find_all(name='tr', attrs={"data-canteen": data_canteen})
+    except AttributeError:
+        # No sides
+        pass
+
+    [parse_dish(dish, canteen)for dish in dishes + sides]
```

Comment:
As you are not returning anything from the function, a simple for loop is more idiomatic:
```suggestion
    for dish in dishes + sides:
        parse_dish(dish, canteen)
```